src/Components/PropertyInspector/PropertyTree/TreeComponents/TreeNodeArrayItemTool.tsx

```diff
@@ -0,0 +1,116 @@
+import { Icon, IIconStyleProps, IIconStyles } from '@fluentui/react';
+import React, { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { PropertyTreeContext } from '../PropertyTree';
+import { NodeProps } from '../PropertyTree.types';
+import TreeNodeSetUnset from './TreeNodeSetUnset';
+
+const TreeNodeArrayItemTool: React.FC<NodeProps> = ({ node }) => {
+    const { t } = useTranslation();
+    const {
+        onRemoveArrayItem,
+        onAddArrayItem,
+        onClearArray,
+        readonly
+    } = useContext(PropertyTreeContext);
+
+    const iconStyles = (props: IIconStyleProps): Partial<IIconStyles> => ({
+        root: {
+            color: props.theme.palette.neutralPrimaryAlt
+        }
+    });
+
+    const handleRemoveArrayItem = () => {
+        onRemoveArrayItem(node);
+    };
+
+    const handleAddArrayItem = () => {
+        onAddArrayItem(node);
+    };
+
+    const handleClearArray = () => {
+        onClearArray(node);
+    };
+    if (readonly) {
+        return null;
+    }
+
+    if (node.schema === 'Array') {
+        const clearButton = (
+            <div
+                tabIndex={0}
+                className={'cb-property-tree-node-map-value-tool-icon'}
+                aria-label={t('propertyInspector.clearArrayIconTitle')}
+                onClick={handleClearArray}
+                onKeyDown={(e) => {
+                    if (e.key === 'Enter') {
+                        handleClearArray();
+                    }
+                }}
+            >
+                <Icon
+                    title={t('propertyInspector.clearArrayIconTitle')}
+                    iconName={'Trash'}
+                    styles={iconStyles}
+                />
+            </div>
+        );
+        const addButton = (
+            <div
+                tabIndex={0}
+                className={'cb-property-tree-node-map-value-tool-icon'}
+                aria-label={t('propertyInspector.addArrayItemIconTitle')}
+                onClick={handleAddArrayItem}
+                onKeyDown={(e) => {
+                    if (e.key === 'Enter') {
+                        handleAddArrayItem();
+                    }
+                }}
+            >
+                <Icon
+                    title={t('propertyInspector.addArrayItemIconTitle')}
+                    iconName={'AddTo'}
+                    styles={iconStyles}
+                />
+            </div>
+        );
+        if (node.children?.length > 0) {
+            return (
+                <>
+                    {addButton}
+                    {clearButton}
+                </>
+            );
+        }
+        return (
+            <>
+                {addButton}
+                <TreeNodeSetUnset node={node} />
+            </>
+        );
+    }
+    if (!node.isArrayItem) return null;
+    return (
```

Comment:
I'm a bit confused why we have 3 paths here, It seems the first is handling the array case, the second is if it's not an array and then I'm wondering how we ever get to the final case. Can we add some comments that explains that for future me?

Reply:
The first (`node.schema === 'Array'`) is for the setting the array root to empty. The final return is for removing the individual array item. I'll reconfigure the conditionals so they make more sense (and put in some comments).